crates/fuel-core/src/database/storage.rs

```diff
@@ -3,32 +3,104 @@ use crate::database::{
     Database,
 };
 use fuel_core_storage::{
-    tables::FuelBlockRoots,
     Error as StorageError,
     Mappable,
     Result as StorageResult,
     StorageInspect,
     StorageMutate,
 };
-use fuel_core_types::blockchain::primitives::BlockHeight;
+use fuel_core_types::{
+    blockchain::primitives::{
+        BlockHeight,
+        BlockId,
+    },
+    fuel_merkle::binary,
+    fuel_types::Bytes32,
+};
 use serde::{
     de::DeserializeOwned,
     Serialize,
 };
 use std::borrow::Cow;
 
-impl DatabaseColumn for FuelBlockRoots {
-    fn column() -> Column {
-        Column::BlockHeaderMerkle
+/// Metadata for dense Merkle trees
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct DenseMerkleMetadata {
+    /// The root hash of the dense Merkle tree structure
+    pub root: Bytes32,
+    /// The version of the dense Merkle tree structure is equal to the number of
+    /// leaves. Every time we append a new leaf to the Merkle tree data set, we
+    /// increment the version number.
+    pub version: u64,
+}
+
+impl Default for DenseMerkleMetadata {
+    fn default() -> Self {
+        let mut empty_merkle_tree = binary::in_memory::MerkleTree::new();
+        Self {
+            root: empty_merkle_tree.root().into(),
+            version: 0,
+        }
     }
 }
 
+/// The table of fuel block's secondary key - `BlockHeight`.
+/// It links the `BlockHeight` to corresponding `BlockId`.
+pub struct FuelBlockSecondaryKeyBlockHeights;
+
+impl Mappable for FuelBlockSecondaryKeyBlockHeights {
+    /// Secondary key - `BlockHeight`.
+    type Key = BlockHeight;
+    type OwnedKey = Self::Key;
+    /// Primary key - `BlockId`.
+    type Value = BlockId;
+    type OwnedValue = Self::Value;
+}
+
+/// The table of BMT MMR data for the fuel blocks.
+pub struct FuelBlockMerkleData;
+
+impl Mappable for FuelBlockMerkleData {
+    type Key = u64;
+    type OwnedKey = Self::Key;
+    type Value = binary::Primitive;
+    type OwnedValue = Self::Value;
+}
+
+/// The metadata table for [`FuelBlockMerkleData`] table.
+pub struct FuelBlockMerkleMetadata;
+
+impl Mappable for FuelBlockMerkleMetadata {
```

Comment:
Is the idea to keep these mappables here instead of fuel-core-storage because they aren't needed anywhere else other than fuel-core right now?